### PR TITLE
chore: BillingProduct cleanup

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

Noticed this bug when visiting the billing page locally

Screenshot 2024-03-12 at 18 54 42

## Changes

- Improve typing
- Account for fact that `currentPlan` could not be present
- Cleanup tooltips on buttons

## How did you test this code?

Verified page loaded locally for me now

### CODE REVIEW

File: frontend/src/types.ts [hunk 1]
```diff
@@ -1470,7 +1470,7 @@ export interface BillingV2PlanType {
     note: string | null
     unit: string | null
     product_key: ProductKeyUnion
-    current_plan?: any
+    current_plan?: boolean
```
Comment (xrdt) [inline frontend/src/types.ts:1473]: Does this work? 

```suggestion
    current_plan?: BillingPlanV2Type | null
```

File: frontend/src/scenes/billing/BillingProduct.tsx [hunk 2]
```diff
@@ -124,9 +124,7 @@ export const BillingProductAddon = ({ addon }: { addon: BillingProductV2AddonTyp
                 </div>
                 <div className="ml-4 mr-4 mt-2 self-center flex gap-x-2 whitespace-nowrap">
                     {addon.docs_url && (
-                        <Tooltip title="Read the docs">
-                            <LemonButton icon={<IconDocument />} size="small" to={addon.docs_url} />
-                        </Tooltip>
+                        <LemonButton icon={<IconDocument />} size="small" to={addon.docs_url} tooltip="Read the docs" />
```
Comment (xrdt) [inline frontend/src/scenes/billing/BillingProduct.tsx:127]: neat that tooltip is now included in buttons.

**Review by xrdt — APPROVED** [review]

File: frontend/src/types.ts [hunk 3]
```diff
@@ -1470,7 +1470,7 @@ export interface BillingV2PlanType {
     note: string | null
     unit: string | null
     product_key: ProductKeyUnion
-    current_plan?: any
+    current_plan?: BillingPlanV2Type | null
```
Comment (daibhin) [inline frontend/src/types.ts:1473]:
```suggestion
    current_plan?: BillingV2PlanType | null
```

**Review by zlwaterfield — APPROVED** [review]
Thanks for making this change!

File: frontend/src/types.ts [hunk 4]
```diff
@@ -1470,7 +1470,7 @@ export interface BillingV2PlanType {
     note: string | null
     unit: string | null
     product_key: ProductKeyUnion
-    current_plan?: any
+    current_plan?: boolean | null
```
Comment (daibhin) [inline frontend/src/types.ts:1473]: Based on `frontend/src/mocks/fixtures/_billing_v2.tsx` and confirming in the Network tab locally this is actually a boolean @xrdt. Just added a new commit and this should be good to go 😄

**Pull request merged**